src/check_rabbitmq.py

```
@@ -0,0 +1,467 @@
+
+    if code == ExitCodes.ok:
+        status = 'OK'
+    elif code == ExitCodes.warning:
+        status = 'WARNING'
+    elif code == ExitCodes.critical:
+        status = 'CRITICAL'
+    else:
+        status = 'UNKNOWN'
+
+    print('{} | {}'.format(status, reason))
+
+    exit(code)
+
+
+class RabbitMQException(Exception):
+    """Exception for reachability of RabbitMQ"""
+    pass
+
+
+class ExitCodes:
+    """Nagios exit codes"""
+
+    ok = 0
+    warning = 1
+    critical = 2
+    unknown = 3
+
+    @staticmethod
+    def get_code(status):
+        """Returns the exit code for a written nagios status"""
+
+        if status == 'ok':
+            return ExitCodes.ok
+        elif status == 'warning':
+            return ExitCodes.warning
+        elif status == 'critical':
+            return ExitCodes.critical
+        elif status == 'unknown':
+            return ExitCodes.unknown
+        else:
+            raise RuntimeError('Status "%s" does not exist' % status)
+
+
+class Check:
+    """Check consists of the variable name, operator, and a value"""
+
+    executors = {
+        '~=': lambda b: re.compile(b).match,
+        '==': lambda b: lambda a: a == b,
+        '!=': lambda b: lambda a: a != b,
+        '<=': lambda b: lambda a: a <= b,
+        '>=': lambda b: lambda a: a >= b,
+        '<': lambda b: lambda a: a < b,
+        '>': lambda b: lambda a: a > b,
+    }
+
+    def __init__(self, key, operator, value):
+        self.key = key
+        self.operator = operator
+        self.value = value
+        self.executor = self.executors[operator](value)
+
+    def __str__(self):
+        return '{} {} {}'.format(self.key, self.operator, self.value)
+
+    def __call__(self, data):
+        """Executes the check itself"""
+
+        name = 'name' in data or 'overview'
+
+        try:
+            for part in self.key.split('.'):
+                data = data[part]
+
+            return self.executor(data), self, name, data
+        except KeyError:
```

I would prefer checking for `if part in data:` over `try: except:`.  Expecting builtin errors is dangerous.  This code can hide the errors from you, if you fatfinger something inside the `executor()` method to cause `KeyError`.